server/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -156,7 +159,13 @@ public Environment(final Settings settings, final Path configPath) {
             assert sharedDataFile != null;
             finalSettings.put(Environment.PATH_SHARED_DATA_SETTING.getKey(), sharedDataFile.toString());
         }
-        this.settings = finalSettings.build();
+
+        SecureSettings secureSettings = finalSettings.getSecureSettings();
+        if (DiscoveryNode.isStateless(settings)) {
+            this.settings = FallbackSecureSettings.installFallbackSecureSettings(finalSettings.build());
```

Comment:
Building the settings requires some extra work. None of the things added to the "final" settings builder are needed in this temporary setup. You can just use `settings`?

Reply:
The final settings are just the `settings` with a couple things added above. I understand a settings builder is needed for setting secure settings, but a builder is not needed to pass into the SecureSettings impl, there you need access to the raw settings. The builder is created inside the install method.

Reply:
OK, I assume from what you say that the couple of added things are probably not necessary at all, so I just pushed a commit that uses the settings. Seems to work!